### PR TITLE
Replace executor with the implementation in go-common-libs

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#7357

### CODE REVIEW

File: app/cmd/nvmecli/nvmecli.go [hunk 1]
```diff
@@ -99,7 +106,8 @@ func ConnectCmd() cli.Command {
 }
 
 func connect(c *cli.Context) error {
-	executor, err := util.GetExecutorByHostProc(c.String("host-proc"))
+	namespaces := []typeslib.Namespace{typeslib.NamespaceMnt, typeslib.NamespaceIpc, typeslib.NamespaceNet}
```
Comment (innobead) [inline app/cmd/nvmecli/nvmecli.go:109]: it: this variable can be a reusable constant, as we always use these three namespaces.

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Just one comment.

File: app/cmd/spdktgt/spdktgt.go [hunk 2]
```diff
@@ -36,5 +38,10 @@ func Cmd() cli.Command {
 }
 
 func spdkTGT(c *cli.Context) error {
-	return target.StartTarget(c.String("spdk-dir"), c.StringSlice("opts"), util.Execute)
+	namespaces := []typeslib.Namespace{typeslib.NamespaceMnt, typeslib.NamespaceIpc, typeslib.NamespaceNet}
+	ne, err := nslib.NewNamespaceExecutor(typeslib.ProcessNone, typeslib.ProcDirectory, namespaces)
+	if err != nil {
+		return err
+	}
+	return target.StartTarget(c.String("spdk-dir"), c.StringSlice("opts"), ne.Execute)
```
Comment (c3y1huang) [inline app/cmd/spdktgt/spdktgt.go:46]: It seems that `util.Execute` was executing commands within the current namespace, which should be achievable with [NewExecutor](https://github.com/longhorn/go-common-libs/blob/main/exec/exec.go#L25). Any reason for opting for `NewNamespaceExecutor`, which involves switching namespaces?

File: pkg/nvme/initiator.go [hunk 3]
```diff
@@ -54,7 +57,8 @@ func NewInitiator(name, subsystemNQN, hostProc string) (*Initiator, error) {
 	}
 
 	// If transportAddress or transportServiceID is empty, the initiator is still valid for stopping
-	executor, err := util.GetExecutorByHostProc("")
+	namespaces := []typeslib.Namespace{typeslib.NamespaceMnt, typeslib.NamespaceIpc, typeslib.NamespaceNet}
+	executor, err := nslib.NewNamespaceExecutor(typeslib.ProcessNone, typeslib.ProcDirectory, namespaces)
```
Comment (c3y1huang) [inline pkg/nvme/initiator.go:61]: Same here, if I'm understanding correctly, `util.GetExecutorByHostProc("")` was performing local execution. Is there any reason for opting namespace switch?

File: app/cmd/spdktgt/spdktgt.go (hunk 2 above)
Comment (derekbit) [reply app/cmd/spdktgt/spdktgt.go:46]: > Any reason for opting for NewNamespaceExecutor, which involves switching namespaces?

The interfaces are mismatching.

> It seems that util.Execute was executing commands within the current namespace, which should be achievable with [NewExecutor](https://github.com/longhorn/go-common-libs/blob/main/exec/exec.go#L25).

I've checked the two namespaced executors and run

- `nsenter --mount=/host/proc/1... --...`
- `nsenter --mount=/proc/1/.... --...`

Frome the results, the first one is executed in in the host namespace, and the second one is in the pod namespace.
It seems the second one is equivalent to NewExecutor.

File: pkg/nvme/initiator.go (hunk 3 above)
Comment (derekbit) [reply pkg/nvme/initiator.go:61]: Explaned in https://github.com/longhorn/go-spdk-helper/pull/56#discussion_r1432466999.
I would like to make the interface clean and work for different namespaces.

File: app/cmd/nvmecli/nvmecli.go [hunk 4]
```diff
@@ -1,9 +1,14 @@
 package nvmecli
 
 import (
+	"fmt"
+
 	"github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
 
+	nslib "github.com/longhorn/go-common-libs/ns"
+	typeslib "github.com/longhorn/go-common-libs/types"
```
[the remark this answers is not in the document]
Comment (derekbit) [reply app/cmd/nvmecli/nvmecli.go]: Addressed. Thank you.

File: app/cmd/nvmecli/nvmecli.go (hunk 1 above)
Comment (derekbit) [reply app/cmd/nvmecli/nvmecli.go]: Addressed. Thank you.

Comment (innobead) [reply app/cmd/nvmecli/nvmecli.go]: @c3y1huang as I commented in another thread, please have this into common-libs as well.

File: app/cmd/spdktgt/spdktgt.go (hunk 2 above)
Comment (innobead) [reply app/cmd/spdktgt/spdktgt.go]: Since PID 1 in the IM pod is a privileged pod, we are able to see all processes there.

**Review by innobead — APPROVED** [review]

**Pull request merged**